File: algorithms/code/leetcode/lc344_reverse_string/lc344_reverse_string.py
# https://leetcode.com/problems/reverse-string/


class Solution:
    def reverseString(self, s: list[str]) -> None:
        for i in range(round(len(s) / 2)):
            s[i], s[len(s) - 1 - i] = s[len(s) - 1 - i], s[i]


class TestCode:
    def test_example(self) -> None:
        s = list("hello")
        Solution().reverseString(s)
        assert list("olleh") == s

File: algorithms/code/leetcode/lc344_reverse_string/lc344_reverse_string.go
// Package lc344 implements https://leetcode.com/problems/reverse-string/
package lc344

func reverseString(s []byte) {
	for i := range len(s) / 2 {
		s[i], s[len(s)-1-i] = s[len(s)-1-i], s[i]
	}
}
